solo-chains/runtime/starlight/src/lib.rs

```diff
@@ -1604,6 +1609,73 @@ parameter_types! {
     pub const MaxLengthParaIds: u32 = 100u32;
     pub const MaxEncodedGenesisDataSize: u32 = 5_000_000u32; // 5MB
 }
+
+pub struct InnerStarlightRegistrar<AccountId, RegistrarManager, RegistrarWeightInfo>(
+    PhantomData<(AccountId, RegistrarManager, RegistrarWeightInfo)>,
+);
+impl<AccountId, RegistrarManager, RegistrarWeightInfo> RegistrarHandler<AccountId>
+    for InnerStarlightRegistrar<AccountId, RegistrarManager, RegistrarWeightInfo>
+where
+    RegistrarManager: RegistrarInterface<AccountId = AccountId>,
+    RegistrarWeightInfo: paras_registrar::WeightInfo,
+{
+    fn register(
+        who: AccountId,
+        id: ParaId,
+        genesis_storage: Vec<ContainerChainGenesisDataItem>,
+        head_data: Option<HeadData>,
+    ) -> DispatchResult {
+        // Return early if head_data is not specified
+        let genesis_head = match head_data {
+            Some(data) => data,
+            None => return Err(DispatchError::Other("HeadData not specified!")),
+        };
+
+        let key_values: Vec<(Vec<u8>, Vec<u8>)> =
+            genesis_storage.into_iter().map(Into::into).collect();
+
+        // Check if the wasm code is present in storage
+        let validation_code = match key_values
+            .into_iter()
+            .find(|(key, _)| key == &StorageWellKnownKeys::CODE.to_vec())
+        {
+            Some((_, code)) => ValidationCode(code),
+            None => return Err(DispatchError::Other("Code not found")),
+        };
+
+        // Try to register the parachain
+        RegistrarManager::register(who, id, genesis_head, validation_code)
+    }
+
+    fn schedule_para_upgrade(id: ParaId) -> DispatchResult {
+        if !RegistrarManager::is_parachain(id) {
+            return RegistrarManager::make_parachain(id);
+        }
+        Ok(())
+    }
+
+    fn schedule_para_downgrade(id: ParaId) -> DispatchResult {
+        if !RegistrarManager::is_parathread(id) {
+            return RegistrarManager::make_parathread(id);
+        }
+        Ok(())
```

Comment:
this means we are not erroring if we schedule a para downgrade for somehitng that is already a parathread right? better document it

Reply:
Done in 9a76e31422088f5a1231e8671063cb0ed3da1581.

Reply:
_fixed_